homeassistant/components/matter/binary_sensor.py

```diff
@@ -95,6 +95,15 @@ def _update_from_device(self) -> None:
         entity_class=MatterBinarySensor,
         required_attributes=(clusters.OccupancySensing.Attributes.Occupancy,),
     ),
+    MatterDiscoverySchema(
+        platform=Platform.BINARY_SENSOR,
+        entity_description=MatterBinarySensorEntityDescription(
+            key="WaterLeakDetector",
+            device_class=BinarySensorDeviceClass.MOISTURE,
+        ),
+        entity_class=MatterBinarySensor,
+        required_attributes=(clusters.BooleanState.Attributes.StateValue,),
```

Comment:
I think you also need to add something like

```

        device_type=(
            device_types.WaterLeakDetector,
        ),
```

otherwise detection will conflict with the door schema higher up, since both are matching on the same attribute (`clusters.BooleanState.Attributes.StateValue`).